src/error.rs

```diff
@@ -38,7 +38,10 @@ pub enum Error {
 
 impl Display for Error {
     fn fmt(&self, fmt: &mut Formatter<'_>) -> fmt::Result {
-        write!(fmt, "{:?}", self)
+        match self {
+            Error::RpcFailure(rpc_failure) => write!(fmt, "RpcFailure({})", rpc_failure),
```

Comment:
"RpcFailure(RpcStatus {...})" seems a little verbose to me. How about just keeping "RpcFailure"?

Reply:
So something like:
`RpcFailure({ status: 13-INTERVAL, details: Some("Something very bad happened") })` ?

I'm not sure just `RpcFailure` is enough because it really nice to have the status in the error message.

I don't really like: `RpcFailure { status: 13-INTERNVAL, details: Some("Something very bad happened") }` because it makes RpcFailure look like a struct when it's a enum. We could also change the RpcFailure variant definition to fit the error:
**From:**
```rust
enum Error {
    RpcFailure(RpcStatus)
}
```

**To** 
```rust
enum Error {
    RpcFailure {
       status: RpcStatusCode,
       code: Option<String>
    }
}
```

But that would be a breaking change.